pkg/virt-api/webhooks/amd64.go

```diff
@@ -0,0 +1,52 @@
+/* Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021
+ *
+ */
+
+/*
+ * arm64 utilities are in the webhooks package because they are used both
+ * by validation and mutation webhooks.
+ */
+package webhooks
+
+import (
+	v1 "kubevirt.io/api/core/v1"
+)
+
+// setDefaultDisksBus set default Disks Bus
+func setAmd64DefaultDisksBus(vmi *v1.VirtualMachineInstance) {
+	// Setting SATA as the default bus since it is typically supported out of the box by
+	// guest operating systems (we support only q35 and therefore IDE is not supported)
+	// TODO: consider making this OS-specific (VIRTIO for linux, SATA for others)
+	bus := v1.DiskBusSATA
+
+	for i := range vmi.Spec.Domain.Devices.Disks {
+		disk := &vmi.Spec.Domain.Devices.Disks[i].DiskDevice
+
+		if disk.Disk != nil && disk.Disk.Bus == "" {
+			disk.Disk.Bus = bus
+		}
+		if disk.CDRom != nil && disk.CDRom.Bus == "" {
+			disk.CDRom.Bus = bus
+		}
+		if disk.LUN != nil && disk.LUN.Bus == "" {
+			disk.LUN.Bus = bus
+		}
+	}
+}
+
+// SetVirtualMachineInstanceAmd64Defaults is mutating function for mutating-webhook
+func SetVirtualMachineInstanceAmd64Defaults(vmi *v1.VirtualMachineInstance) {
```

Comment:
Would it make sense to move the call of `SetObjectDefaults_VirtualMachineInstance` into the bottom ofthe arch-specific defaulters? Basically ensuring that we always only have to call the arch-specific defaulter and we are good?

Reply:
Do you means, in the `Mutate( )` function, we do arch specific default setting first then do `SetObjectDefaults_VirtualMachineInstance` ?
```
                if webhooks.IsARM64() {
                        log.Log.V(4).Info("Apply Arm64 specific setting")
                        webhooks.SetVirtualMachineInstanceArm64Defaults(newVMI)
                } else {
                        webhooks.SetVirtualMachineInstanceAmd64Defaults(newVMI)
                        mutator.setDefaultCPUModel(newVMI)
                }
                v1.SetObjectDefaults_VirtualMachineInstance(newVMI)
```

Reply:
Yes, what I just find interesting, we have this comment:

```
// Do some specific setting for Arm64 Arch. It should put before SetObjectDefaults_VirtualMachineInstance
```

but we actually call `SetObjectDefaults_VirtualMachineInstance` before it.

I am fine with the current flow too, for as long as we don't  override defaults in arch specific settings.

Reply:
Let's just ensure that we sort out the difference between the code flow and the comment.

Reply:
I just take a look if how we can put arch specific defaulter before `SetObjectDefaults_VirtualMachineInstance`. It seems no conflict, and I think your logic is right. I will update it.